config.md

```diff
@@ -1,3 +1,455 @@
-# TranslucentTB Config documentation
+# TranslucentTB Configuration Documentation
 
-TODO: write documentation
\ No newline at end of file
+TranslucentTB uses a JSON configuration file to customize the appearance and behavior of the Windows taskbar. This document explains all available configuration options and their usage.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Configuration Structure](#configuration-structure)
+- [Appearance Settings](#appearance-settings)
+- [Appearance Modes](#appearance-modes)
+- [Window Rules](#window-rules)
+- [Global Settings](#global-settings)
+- [Example Configurations](#example-configurations)
+- [Tips and Best Practices](#tips-and-best-practices)
+- [Troubleshooting](#troubleshooting)
+
+## Overview
+
+### Configuration File Location
+
+The configuration file should be saved as `settings.json` in the TranslucentTB application directory or user data folder.
+
+### Schema Reference
+
+The configuration follows the JSON schema available at: `https://TranslucentTB.github.io/settings.schema.json`
+
+## Configuration Structure
+
+```json
+{
+  "$schema": "https://TranslucentTB.github.io/settings.schema.json",
+  "desktop_appearance": { ... },
+  "visible_window_appearance": { ... },
+  "maximized_window_appearance": { ... },
+  "start_opened_appearance": { ... },
+  "search_opened_appearance": { ... },
+  "task_view_opened_appearance": { ... },
+  "battery_saver_appearance": { ... },
+  "ignored_windows": { ... },
+  "hide_tray": false,
+  "disable_saving": false,
+  "verbosity": "warn",
+  "language": "en-US",
+  "use_xaml_context_menu": false
+}
+```
+
+## Appearance Settings
+
+All appearance configurations support the following properties:
+
+### Basic Properties
+
+#### `accent` (string)
+The accent/transparency effect to apply:
+- `"normal"`: Windows default appearance
+- `"opaque"`: Solid, non-transparent taskbar
+- `"clear"`: Fully transparent taskbar
+- `"blur"`: Blurred transparency effect
+- `"acrylic"`: Windows 10/11 acrylic material effect
+
+#### `color` (string)
+Custom color in hexadecimal format:
+- Format: `#RRGGBB` or `#RRGGBBAA` (with alpha channel)
+- Example: `"#00000000"` (fully transparent black)
+
+#### `show_peek` (boolean)
+Enable/disable desktop peek functionality:
+- `true`: Show desktop peek
+- `false`: Disable desktop peek
+
+#### `show_line` (boolean)
+Show/hide the taskbar accent line:
+- `true`: Display accent line
+- `false`: Hide accent line
+
+#### `blur_radius` (number)
+Blur effect intensity (when using blur accent):
```

Comment:
Worth documenting that this only works with the modern XAML-based taskbar in Windows 11